js/index_controls.js

```diff
@@ -79,8 +79,28 @@ function calculate() {
 				: Math.floor(p2.maxHP * p1.moves[i].hasRecoil * 0.48 / p1.maxHP);
 			}
 			recoilText = ' (' + minRecoilDamage + ' - ' + maxRecoilDamage + notation + ' recoil damage)';
+		} else if (p1.moves[i].hasRecoil === 'crash') {
+			var minRecoilDamage = notation === '%' ? Math.floor(Math.min(minDamage, p2.maxHP) * 10 / p1.maxHP) :
+				Math.floor(Math.min(minDamage, p2.maxHP) * 0.48 / p1.maxHP);
+			var maxRecoilDamage = notation === '%' ? Math.floor(Math.min(maxDamage, p2.maxHP) * 10 / p1.maxHP) :
+				Math.floor(Math.min(maxDamage, p2.maxHP) * 0.48 / p1.maxHP);
+			if (minDamage > p2.curHP) {
+				minRecoilDamage = notation === '%' ? Math.floor(p2.maxHP * 10 / p1.maxHP) / 10
+				: Math.floor(p2.maxHP * 0.48 / p1.maxHP);
+				maxRecoilDamage = notation === '%' ? Math.floor(p2.maxHP * p1.moves[i].hasRecoil * 10 / p1.maxHP) / 10
+				: Math.floor(p2.maxHP * 0.48 / p1.maxHP);
+			}
+			var gen4CrashDamage = Math.floor(p2.maxHP * 0.5 / p1.maxHP * 100);
+			recoilText = gen === 1 ? ' (1hp damage on miss)' :
+			gen === 2 ? (p2.type1 === "Ghost" || p2.type2 === "Ghost") ? ' (no crash damage on Ghost types)' : ' (' + minRecoilDamage * 12.5 + ' - ' + maxRecoilDamage * 12.5 + notation + ' crash damage)' :
```

Comment:
Gen 2's crash damage isn't capped at the target's max HP. If you're level 100 and you miss against something that's level 2, you probably just KOed yourself.

Reply:
Oh that’s funny, I’ll remove the cap